src/python/tests/core/test_scheduler.py

```diff
@@ -9,25 +9,119 @@
+_STORAGE_DIR = os.path.expanduser(_FOGLAMP_ROOT + '/services/storage')
+
+"""
+    _is_management_started, _address, _host, _m_port, _app, _server_handler, _server are module level variables.
+
+    start_storage(), stop_storage() and start_management() are module level functions.
+
+    setup_module() and teardown_module() are module level setup and teardown methods but we are using teardown_module()
+    only.
+
+    setup has been moved into each individual test as start_management(), essential part of setup, is a coro and is
+    called by the event loop and setup_module() is never called by the event loop. Also, though setup has been defined
+    in each test, as we do not know the order of the execution of the tests, BUT is executed only once.
+"""
+
+_is_management_started = False
+_address = None
+_host = '0.0.0.0'
+_m_port = 0
+_app = None
+_server_handler = None
+_server = None
+
+def start_storage(host, m_port):
+    try:
+        cmd_with_args = ['./storage', '--address={}'.format(host),
+                         '--port={}'.format(m_port)]
+        import subprocess
+        subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+    except Exception as ex:
+        pass
+
+def stop_storage():
+    try:
+        Storage().shutdown()
+    except Exception as ex:
+        pass
+
+async def start_management():
+    global _is_management_started, _address, _host, _m_port, _app, _server_handler, _server
+
+    loop = asyncio.get_event_loop()
+    _app = Server._make_core_app()
+    _server_handler = _app.make_handler()
+    coro = loop.create_server(_server_handler, _host, 0)
+    # added coroutine
+    _server = await coro
+    _address, _m_port = _server.sockets[0].getsockname()
+
+def setup_module():
+    pass
+
+def teardown_module():
+    global _is_management_started, _address, _host, _m_port, _app, _server_handler, _server
+
+    # Stop Storage
+    stop_storage()
+    # Stop Management
+    loop = asyncio.get_event_loop()
+    _server.close()
+    loop.run_until_complete(_server.wait_closed())
+    loop.run_until_complete(_app.shutdown())
+
 
 @pytest.allure.feature("unit")
 @pytest.allure.story("scheduler")
 class TestScheduler:
+    _engine = None  # type: aiopg.sa.Engine
+
+    async def _get_connection_pool(self) -> aiopg.sa.Engine:
+        """Returns a database connection pool object"""
+        if self._engine is None:
+            self._engine = await aiopg.sa.create_engine(_CONNECTION_STRING)
+        return self._engine
+
+    async def populate_test_data(self):
+        """Delete all schedule-related tables and insert processes for testing"""
+        async with (await self._get_connection_pool()).acquire() as conn:
+            await conn.execute('delete from foglamp.tasks')
+            await conn.execute('delete from foglamp.schedules')
+            await conn.execute('delete from foglamp.scheduled_processes')
+            await conn.execute(
+                '''insert into foglamp.scheduled_processes(name, script)
```

Comment:
We can't call sleep anymore, as noted in the scrum call. I suggest we wrap the sleep in a shell script, that way we can also test the scheduler is passing the three required arguments --name=, --address= and --port=